### PR TITLE
feat: Adds custom max concurrent request on jsonrpsee http client

### PR DESCRIPTION
**Pull request by moraesjeremias**
# Problem Statement
jsonrpsee lib has a default 256 maximum concurrent rpc requests on its http client. See reference below.
In load test scenarios we caught 20k errors due to maximum concurrent slots field in the HTTP client.

## Proposal
- Introduce a configurable/dynamic max_concurrent_request var to correctly build/construct the HTTP client with accordingly specs.

## To be tested
- [ ] Load with customized max concurrent request values (256, 512, 1024)
- [ ] Resource consumption on higher max concurrent request values.

## Reference
[jsonrpsee lib HttpClientBuilder implementation.](https://github.com/paritytech/jsonrpsee/blob/257513ede6de5e4242b49d0089a606c59fa2d29d/client/http-client/src/client.rs#L237)
```rust
impl Default for HttpClientBuilder<Identity> {
	fn default() -> Self {
		Self {
			max_request_size: TEN_MB_SIZE_BYTES,
			max_response_size: TEN_MB_SIZE_BYTES,
			request_timeout: Duration::from_secs(60),
			max_concurrent_requests: 256,
			certificate_store: CertificateStore::Native,
			id_kind: IdKind::Number,
			max_log_length: 4096,
			headers: HeaderMap::new(),
			service_builder: tower::ServiceBuilder::new(),
			tcp_no_delay: true,
		}
	}
}
```

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -585,14 +585,21 @@ async fn main() {
...
         .init();
 
+    let max_concurrent_request: usize = env::var("MAX_CONCURRENT_REQUEST")
+        .ok()
+        .and_then(|max_request| max_request.parse::<usize>().ok())
+        .unwrap_or(256);
+
     let shared_state = Arc::new(AppState {
         avail_client: HttpClientBuilder::default()
+            .max_concurrent_requests(max_concurrent_request)
```
Comment (QEDK) [inline src/main.rs:595]: Can be changed to `1024` requests.

**Review by QEDK — CHANGES_REQUESTED** [review]
Nit.

File: src/main.rs (hunk 1 above)
Comment (moraesjeremias) [reply src/main.rs:630]: Done :white_check_mark:

**Review by QEDK — APPROVED** [review]
LGTM.

**Pull request merged**